.github/workflows/CI.yml

```diff
@@ -0,0 +1,69 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+      - uses: pre-commit/action@2c7b3805fd2a0fd8c1884dcaebf91fc102a13ecd #v3.0.1
+
+  pypi-build:
+    name: Build package for PyPI
+    if: github.repository == 'access-nri/model-config-tests'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Python 3.11
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+          cache: 'pip'
+
+      - name: Install build dependencies
+        run: python3 -m pip install --upgrade build
+
+      - name: Build distributions
+        run: python3 -m build
+
+      - name: Upload distribution artifact
+        uses: actions/upload-artifact@v4
+        with:
+          path: dist
```

Comment:
We upload this artifact as part of CI, but we don't download it again during this run. Just making sure that we don't want to use it for anything in the later `tests` job. It's fine to just have it for provenance, but remember that workflow artifacts have a 90 day expiry.

Reply:
I think it's mostly there to test the `pypi-build` job in CD will work fine, so I could remove the upload part as probably it is unnecessary to store all these artefacts for 90 days.